provision/acc_provision/acc_provision.py

```diff
@@ -314,14 +314,15 @@ def config_user(config_file):
                 data = file.read()
         user_input = re.sub('password:.*', '', data)
         config["user_input"] = user_input
-        if not isinstance(config["net_config"]["pod_subnet"], list):
-            config["net_config"]["pod_subnet"] = [config["net_config"]["pod_subnet"]]
+        if flavor != "l2-tenant-chained-multus-openshift-4.12-baremetal":
```

Comment:
Rather than creating a completely different flavor, can we investigate if we can add a configuration option to indicate chained CNI mode? We do something similar in the case of CKO. Otherwise the number of flavors, considering the distros and the combinations therein, can blow up. Ideally it would be good to have one flavor for each distro (and we already have that flavor) and process it differently based on the options selected.

Reply:
Hi Sumit, what networking the chained cni needs to provide, depends on how the primary CNI is configured. If it uses the external peering method of calico/cilium then chained CNI need not do anything for networking.  This is indeed a special case where OVN is doing overlay over ACI and chained CNI is handling additional networks only. We can name it accordingly.

Reply:
Sumit, a flavor for ovn-openshift-baremetal or overlay-openshift-baremetal is still required. chained can be a flag as you mention.

Reply:
Okay that makes sense. Let's include the official name "openshift-sdn-ovn-baremetal". But I do have a separate question, are we going to have a separate flavor for the chained plugin for the additional interfaces (SR-IOV, MacVLAN,..)?

Reply:
No, that is read based on the NAD, so not a static config.

Reply:
So we are dynamically inserting the chained CNI for the additional interfaces?

Reply:
NAD is created by the user. We can have him annotate it and edit to add a chain based on that. I am not doing this right now, but if its ok editing something that the user has created, I can do this change.

Reply:
No, we shouldn't require the user to create the NAD in a way different they are currently doing (that could be an option but not the only option). In a given deployment the intent to add a chained CNI is a "meta" intent. To signal this intent and any configuration that goes with it, it seemed natural to have a flavor for the corresponding secondary CNI.